game-core/src/main/java/games/strategy/triplea/ui/UnitIconProperties.java

```diff
@@ -0,0 +1,114 @@
+package games.strategy.triplea.ui;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.triplea.attachments.AbstractConditionsAttachment;
+import games.strategy.triplea.attachments.AbstractPlayerRulesAttachment;
+import games.strategy.triplea.attachments.ICondition;
+import games.strategy.util.FileNameUtils;
+
+/**
+ * Loads objective text from objectives.properties.
+ */
+public class UnitIconProperties extends PropertyFile {
+
+  private static final String PROPERTY_FILE = "unit_icons.properties";
+
+  private static UnitIconProperties instance = null;
+  private static Instant timestamp = Instant.EPOCH;
+  private static Map<ICondition, Boolean> conditionsStatus = new HashMap<>();
+
+  protected UnitIconProperties(final GameData data) {
+    super(PROPERTY_FILE);
+    final String gameName =
+        FileNameUtils.replaceIllegalCharacters(data.getGameName(), '_').replaceAll(" ", "_").concat(".");
+    for (final String key : properties.stringPropertyNames()) {
+      if (!key.startsWith(gameName)) {
+        continue;
+      }
+      try {
+        final String[] unitInfoAndCondition = key.split(";");
+        final String[] unitInfo = unitInfoAndCondition[0].split("\\.");
+        if (unitInfoAndCondition.length != 2) {
+          continue;
+        }
+        final ICondition condition =
+            AbstractPlayerRulesAttachment.getCondition(unitInfo[1], unitInfoAndCondition[1], data);
+        if (condition != null) {
+          conditionsStatus.put(condition, false);
+        }
+      } catch (final Exception e) {
+        System.err.println("unit_icons.properties keys must be: <game_name>.<player>.<unit_type>;attachmentName OR "
+            + "if always true: <game_name>.<player>.<unit_type>");
+      }
+    }
+    conditionsStatus = getTestedConditions(data);
+  }
+
+  public static UnitIconProperties getInstance(final GameData data) {
+    // cache properties for 10 seconds
+    if (instance == null || timestamp.plusSeconds(10).isBefore(Instant.now())) {
+      instance = new UnitIconProperties(data);
+      timestamp = Instant.now();
+    }
+    return instance;
```

Comment:
Just noticed this method is static.
Of course it would be also an option to just drop this caching mechanism instead 1 second isn't really much time and if there are many calls in a short amount of time, we might just want to pass an object somewhere to be used again. 
This also allows the garbage collector to collect this object when it's no longer needed.

Reply:
Yeah, given it was static, I left this properties caching stuff alone for now. I think there are potentially a lot of calls and if you read the properties file every time it could negatively impact performance for some of these. I actually considered just loading the properties file once as I don't think many users change properties while a game is running. Either way I think it isn't that big of a deal to have a few classes with a small caching method duplicated and memory wise its tiny as these are just properties files with at most a few hundred lines.

Reply:
I don't have a problem with the caching itself, I just think holding on to that object reference until the next call to that method, which might never happen is some sort of memory leak.
And bad for tge garbage collector as well depending on how long the reference is being hold.

Reply:
@ron-murhammer #3456